xtable-hive-metastore/src/main/java/org/apache/xtable/hms/table/HudiHMSCatalogTableBuilder.java

```diff
@@ -0,0 +1,207 @@
+      Optional<HoodieTableMetaClient> metaClientOpt =
+          hudiTableManager.loadTableMetaClientIfExists(basePath);
+
+      if (!metaClientOpt.isPresent()) {
+        throw new CatalogSyncException(
+            "failed to get meta client since table is not present in the base path " + basePath);
+      }
+
+      metaClient = metaClientOpt.get();
+    }
+    return metaClient;
+  }
+
+  @Override
+  public Table getCreateTableRequest(
+      InternalTable table, CatalogTableIdentifier catalogTableIdentifier) {
+    HierarchicalTableIdentifier tableIdentifier =
+        toHierarchicalTableIdentifier(catalogTableIdentifier);
+    Table newTb = new Table();
+    newTb.setDbName(tableIdentifier.getDatabaseName());
+    newTb.setTableName(tableIdentifier.getTableName());
+    try {
+      newTb.setOwner(UserGroupInformation.getCurrentUser().getShortUserName());
+    } catch (IOException e) {
+      throw new CatalogSyncException(
+          "Failed to set owner for hms table: " + tableIdentifier.getTableName(), e);
+    }
+
+    newTb.setCreateTime((int) ZonedDateTime.now().toEpochSecond());
+    List<String> partitionFields =
+        table.getPartitioningFields().stream()
+            .map(field -> field.getSourceField().getName())
+            .collect(Collectors.toList());
+    Map<String, String> tableProperties =
+        getTableProperties(partitionFields, table.getReadSchema());
+    newTb.setParameters(tableProperties);
+    newTb.setSd(getStorageDescriptor(table));
+    newTb.setPartitionKeys(getSchemaPartitionKeys(table));
+    return newTb;
+  }
+
+  @Override
+  public Table getUpdateTableRequest(
+      InternalTable table, Table hmsTable, CatalogTableIdentifier tableIdentifier) {
+    Map<String, String> parameters = hmsTable.getParameters();
+    List<String> partitionFields =
+        table.getPartitioningFields().stream()
+            .map(field -> field.getSourceField().getName())
+            .collect(Collectors.toList());
+    Map<String, String> tableParameters = hmsTable.getParameters();
+    tableParameters.putAll(getTableProperties(partitionFields, table.getReadSchema()));
+    hmsTable.setParameters(tableParameters);
+    hmsTable.setSd(getStorageDescriptor(table));
+
+    hmsTable.setParameters(parameters);
+    hmsTable.getSd().setCols(schemaExtractor.toColumns(TableFormat.HUDI, table.getReadSchema()));
+    return hmsTable;
+  }
+
+  private Map<String, String> getTableProperties(
+      List<String> partitionFields, InternalSchema schema) {
+    Map<String, String> tableProperties = new HashMap<>();
+    tableProperties.put(HUDI_METADATA_CONFIG, "true");
+    Map<String, String> sparkTableProperties =
+        HudiCatalogTableUtils.getSparkTableProperties(
+            partitionFields, "", hmsCatalogConfig.getSchemaLengthThreshold(), schema);
+    tableProperties.putAll(sparkTableProperties);
+    return tableProperties;
+  }
+
+  @VisibleForTesting
+  StorageDescriptor getStorageDescriptor(InternalTable table) {
```

Comment:
This method is only called from within this class, should this method move back to `private` scope and the annotation removed?